### ISSUE TITLE
A schema's field cannot be of type List[str]

### ISSUE THREAD

**Issue by nicolas-leydet** [Bogdanp/molten#12]
## What OS are you using?
SLED 12.3

## What version of molten are you using?
0.50

## What did you do?
Use the following Schema as an handler return annotation  
```python
@schema
class SearchResult:
    results: List[str]
```

## What did you expect would happen?
No error

## What happened?
```bash  
File "~/.virtualenvs/matching/lib/python3.7/site-packages/molten/openapi/documents.py", line 239, in generate_openapi_document
    response_schema_name = _generate_schema("response", response_annotation, schemas)
  File "~/.virtualenvs/matching/lib/python3.7/site-packages/molten/openapi/documents.py", line 319, in _generate_schema
    is_optional, field_schema = _generate_field_schema(field, context, schemas)
  File "~/.virtualenvs/matching/lib/python3.7/site-packages/molten/openapi/documents.py", line 345, in _generate_field_schema
    item_schema_name = _generate_schema(context, arguments[0], schemas)
  File "~/.virtualenvs/matching/lib/python3.7/site-packages/molten/openapi/documents.py", line 303, in _generate_schema
    for field in schema._FIELDS.values():  # noqa
AttributeError: 'function' object has no attribute '_FIELDS'
```
[this](https://github.com/Bogdanp/molten/blob/f25ebb535c45f4de64568a07fea9795179184acb/molten/openapi/documents.py#L345) is not valid if you use anything else than a schema as `List` argument

**Comment by Bogdanp** [comment]
Thanks! I'll take a look at this this weekend.

**Comment by Bogdanp** [comment]
Thanks again for reporting this! I've released a fix in version `0.5.1`.
